x-pack/solutions/observability/plugins/synthetics/public/apps/locators/group_monitor_by_location.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { syntheticsMonitorLocationQueryLocatorID } from '@kbn/observability-plugin/common';
+
+async function navigate({ locationId }: { locationId: string }) {
+  return {
+    app: 'synthetics',
+    path: `?locations=${[locationId]}`,
+    state: {},
+  };
+}
```

Comment:
Would that solve the URL consistency between using the locator and the synthetics UI? But if it works without encoding the array, feel free to ignore this

```suggestion
async function navigate({ locationId }: { locationId: string }) {
  const locations = encodeURIComponent(JSON.stringify([locationId]))

  return {
    app: 'synthetics',
    path: `?locations=${locations}`,
    state: {},
  };
}
```